Release notes for v6.0.0-RC1 (#28901)
containers/podman: RELEASE_NOTES.md

# Release Notes

## 6.0.0
### Breaking Changes

> - Support for BoltDB databases has been dropped. Starting Podman 6 when using BoltDB will attempt an automatic migration from SQLite to BoltDB.

Reviewer: ```suggestion
- Support for BoltDB databases has been dropped. Starting with Podman 6, when using BoltDB, it will attempt an automatic migration from SQLite to BoltDB.
```

- Support for running on Intel Macs has been removed.
- Support for running on Windows 10 has been removed.
- Support for running on cgroups v1 systems has been removed. Please update your system to use cgroups v2.
- Support for running on iptables has been removed. Please use nftables instead.
- Support for CNI networking has been removed. Please use Netavark instead.
- Support for the slirp4netns rootless network stack has been removed. Please use Pasta instead. As part of this, the `--network-cmd-path` global option, only used with `slirp4netns`, has been removed.

> - Podman's import path has changed from `github.com/containers/podman/v5` to `go.podman.io/podman/v6` as part of our move into a CNCF-owned Github organization.

Reviewer: ```suggestion
- Podman's import path has changed from `github.com/containers/podman/v5` to `go.podman.io/podman/v6` as part of our move into a CNCF-owned GitHub organization.
```

- Network isolation now defaults to enabled, improving Docker compatibility and security. A special workaround for the Docker-compatible API related to isolation being disabled has been removed ([#27349](https://github.com/podman-container-tools/podman/issues/27349)).

> - Podman's configuration file parsing logic has seen a major rewrite. Please see [this document](https://github.com/podman-container-tools/podman/blob/main/contrib/design-docs/config-file-parsing.md) for exact details.

Reviewer: I'd be tempted to make this the first line after the "Support for X" has been dropped lines at 11.5.

> - The way the `podman quadlet` suite of commands functions has been changed. Previously, Quadlets and associated files were tracked using a `.app` file, ensuring that removing a Quadlet removed all associated non-Quadlet files. Now, Quadlets and associated files are placed in subdirectories, which should reduce bugs and make manual management of Quadlets added by `podman quadlet install` much easier.

Reviewer: ```suggestion
- The way the `podman quadlet` suite of commands functions has been changed. Previously, Quadlets and their associated files were tracked via a `.app` file, ensuring that removing a Quadlet also removed all associated non-Quadlet files. Now, Quadlets and associated files are placed in subdirectories, which should reduce bugs and make manual management of Quadlets added by `podman quadlet install` much easier.
```

- VMs made by `podman machine` on Linux now mount volumes from the host using systemd. Volume mounts on existing `podman machine` VMs on Linux have been broken by this change, and the VM will need to be recreated.
- The `podman volume prune` command now matches Docker's behavior by only pruning unused anonymous volumes. Please use the newly-added `--all` option for the previous behavior (pruning all volumes).

> - The `podman volume list` command now combines multiple filters using logical AND instead of logical OR (meaning all filters must match for a container to be included in output) ([#26786](https://github.com/podman-container-tools/podman/issues/26786)).

Reviewer: Maybe?
```suggestion
- The `podman volume list` command now combines multiple filters using logical `AND` instead of logical `OR` (meaning all filters must match for a container to be included in output) ([#26786](https://github.com/podman-container-tools/podman/issues/26786)).
```

> - The `label!=` filter used in many commands now combines the output of multiple instances of the filter with logical AND instead of logical OR.

Reviewer: if you do backticks for the prior, do them here and elsewhere.  Also, please search for "Github" through out and change it to "GitHub".

- A new command has been added, `podman machine os update`, which updates the operating system of a `podman machine` VM. Please note that this is not supported with the `wsl` provider.
- A new command has been added, `podman system hyperv-prep`, allowing Windows administrators to prepare a host for their users to run `podman machine` VMs using the `hyperv` provider.
- When starting a VM with `podman machine start` and `podman machine init --now`, if the connection to that VM is not the default, users will be prompted whether they want to change the default to the machine that was just started. This can also be controlled by a new option, `--update-connection`, which controls whether the default will be updated. If the `--update-connection` option is set, a user-interactive prompt is not displayed.

> - The `podman machine init` and `podman machine set` commands now support a new option, `--import-native-ca`, which (when set) causes `podman machine` VMs on Windows, Linux, and Mac to import the host's trusted CA certificates each time the VM boots.

Reviewer: ```suggestion
- The `podman machine init` and `podman machine set` commands now support a new option, `--import-native-ca`, which, when set, causes `podman machine` VMs on Windows, Linux, and Mac to import the host's trusted CA certificates each time the VM boots.
```
Suggestion, and just my preference to avoid parens

- Add support for blackhole, unreachable, and prohibit route types in podman networks. Supported since netavark 2.0.

### Changes
- VMs created by `podman machine` now mount the host's user configurations (e.g. `~/.config/containers` on Linux) into the machine at `/etc/containers`, allowing users to edit the config files controlling Podman's behavior directly.
- The default `podman machine` provider on Macs has been changed to `libkrun`.
- Starting and stopping `podman machine` VMs on Windows with the `hyperv` provider no longer requires administrator privileges (creating machines still requires admin, however). Operations requiring elevated privileges will prompt for administrator access. Please note that this only works with newly-created VMs.
- The `podman pod inspect` command now prints arrays in its output in deterministic order.
- The `podman machine os apply` command has been updated, and now uses `bootc switch` to apply changes. All transports supported by `bootc switch` can be used for the new image to apply.

> - An experimental feature has been added where, on systems using Kernel 6.18 and newer, rootless Podman will no longer needs to create a pause process to hold open the rootless user namespace, instead using an `nsfs` file handle. This behavior is currently gated behind an environment variable, `drop-pause-process`, being set.

Reviewer: ```suggestion
- An experimental feature has been added where, on systems using Kernel 6.18 and newer, rootless Podman will no longer need to create a pause process to hold open the rootless user namespace, instead using an `nsfs` file handle. This behavior is currently gated behind an environment variable, `drop-pause-process`, being set.
```

- Fixed a bug where setting `Entrypoint=""` in a Quadlet `.container` file did not clear the container's entrypoint ([#28213](https://github.com/containers/podman/issues/28213)).
- Fixed a bug where setting a `HealthCmd` in a Quadlet `.container` file to a command that included double-quotes (`"`) would result in a nonfunctional healthcheck due to a parsing issue ([#28409](https://github.com/containers/podman/issues/28409)).
- Fixed a bug where FreeBSD systems could panic when inspecting containers created with the `host` network mode ([#28289](https://github.com/containers/podman/issues/28289)).

### API
- Fixed a bug where the Libpod System Check endpoint could perform operations with bad data after returning a 400 error ([#28350](https://github.com/containers/podman/issues/28350)).
- Fixed a bug where the remote attach API for containers (Libpod & Compat) could panic due to a rare race condition ([#28277](https://github.com/containers/podman/issues/28277)).
- Fixed a bug where the Secret Create API could not create functional secrets using the `shell` driver due to options from the default driver being improperly added.

### Misc

> - Updated Buildah to v1.43.1

Reviewer: ```suggestion
- Updated Buildah to v1.44.0
```

Author: You're onto the Podman 5.8 section

> - Updated the containers/common library to v0.67.1

Reviewer: ```suggestion
- Updated the containers/common library to v0.68.0
```

> - Updated the containers/image library to v5.39.2

Reviewer: ```suggestion
- Updated the containers/image library to v5.40.0
```
For these three, I'm assuming that you're pulling from the podman-6.0 branch

# Release Notes

## 6.0.0
### Breaking Changes
- Support for BoltDB databases has been dropped. Starting Podman 6 when using BoltDB will attempt an automatic migration from SQLite to BoltDB.
- Support for running on Intel Macs has been removed.
- Support for running on Windows 10 has been removed.

> - Support for running on cgroups v1 systems has been removed. Please update to cgroups v2.

Reviewer 2: ```suggestion
- Support for running on cgroups v1 systems has been removed. Please update your system to use cgroups v2.
```

- Events now include artifact lifecycle events (`create`, `pull`, `push`, and `remove`) ([#27260](https://github.com/containers/podman/issues/27260)).
- A new experimental option for the `rootless_port_forwarder` field in `containers.conf` has been added, `rootless_port_forwarder="pasta"`. When set, rootless bridge networks will use Pasta's kernel-level port forwarding via Pesto instead of rootlessport, preserving the original client source IP in network traffic in rootless containers. The default remains `rootlessport` (the default for Podman 5.x), but we will investigate switching at a later date when stability is more certain.
- A new filter has been added to the `podman ps` and `podman container prune` commands, `--filter annotation=`, to filter containers based on their annotations ([#28562](https://github.com/podman-container-tools/podman/issues/28562)).
- The `podman network create` command's `--route` option can now create blackhole, unreachable, and prohibit routes to prevent containers from reaching certain networks (e.g. `podman network create --route 10.20.30.40/24,blackhole ...`) ([#20022](https://github.com/podman-container-tools/podman/issues/20222)).

> 

Reviewer 2: I guess the empty line here should be dropped?

- Add support for blackhole, unreachable, and prohibit route types in podman networks. Supported since netavark 2.0.

### Changes
- VMs created by `podman machine` now mount the host's user configurations (e.g. `~/.config/containers` on Linux) into the machine at `/etc/containers`, allowing users to edit the config files controlling Podman's behavior directly.
- The default `podman machine` provider on Macs has been changed to `libkrun`.
- Starting and stopping `podman machine` VMs on Windows with the `hyperv` provider no longer requires administrator privileges (creating machines still requires admin, however). Operations requiring elevated privileges will prompt for administrator access. Please note that this only works with newly-created VMs.
- The `podman pod inspect` command now prints arrays in its output in deterministic order.
- The `podman machine os apply` command has been updated, and now uses `bootc switch` to apply changes. All transports supported by `bootc switch` can be used for the new image to apply.

> - On systems using Kernel 6.18 and newer, rootless Podman will no longer create a pause process to hold open the rootless user namespace, instead using an `nsfs` file handle.

Reviewer 2: That is not true, we still default to the pause process, there is an opt in env.

- Fixed a bug where the Compat Info endpoint did not return the location of the Seccomp profile if a non-default profile was in use ([#28379](https://github.com/podman-container-tools/podman/issues/28379)).
- Fixed a bug where the Compat List endpoint for Containers could return an invalid string for container status ([#28359](https://github.com/podman-container-tools/podman/issues/28359)).
- Fixed a bug where the Compat List endpoint for Containers did not include the `HostConfig` field in its responses.
- Fixed a bug where the Compat Wait endpoint for Containers would hang indefinitely when waiting for the `next-exit` condition ([#28514](https://github.com/containers/podman/issues/28514)).
- Fixed a bug where the Compat and Libpod Update endpoints for Containers would clear the rlimits of the container if they were not explicitly set in the API request.
- Fixed a bug where the Compat Push endpoint for Images did not return a final JSON object including tag, digest, and size of the pushed image, as Docker does.

### Misc

> - Autocorrect has been enabled for inspecting artifacts with `podman inspect`.

Reviewer 2: What is autocorrect?

Author: Autocomplete. Oops.

Reviewer 2: I guess I would put it under features not misc then but I guess it does not matter so I do not mind either way
